### PR TITLE
feat: Add support for uploading media in admin

### PR DESCRIPTION
**Pull request by sradevski**
Note: Uploading media during creation is not plugged in yet, so I decided to open this PR first, as the creation flow will also require some refactors.

### CODE REVIEW

File: packages/modules/providers/file-s3/src/services/s3-file.ts [hunk 1]
```diff
@@ -99,6 +99,7 @@ export class S3FileService extends AbstractFileProviderService {
       // protected private_secret_access_key_: string
 
       // ACL: options.acl ?? (options.isProtected ? "private" : "public-read"),
+      ACL: "public-read",
```
Comment (carlos-r-l-rodrigues) [inline packages/modules/providers/file-s3/src/services/s3-file.ts:102]: should this be `options.acl ?? (options.isProtected ? "private" : "public-read")` ?

**Review by carlos-r-l-rodrigues — APPROVED** [review]

**Review by olivermrbl — APPROVED** [review]

File: packages/admin-next/dashboard/src/routes/products/product-media/components/edit-product-media-form/edit-product-media-form.tsx [hunk 2]
```diff
@@ -85,20 +86,17 @@ export const EditProductMediaForm = ({ product }: ProductMediaViewProps) => {
...
+        .create({ files })
+        .then((res) => {
+          return res.files
+        })
+        .catch(() => {
+          form.setError("media", {
+            type: "invalid_file",
+            message: t("products.media.failedToUpload"),
+          })
+        })
```
Comment (adrien2p) [inline packages/admin-next/dashboard/src/routes/products/product-media/components/edit-product-media-form/edit-product-media-form.tsx:99]:
```suggestion
      const { files: uploads } = await sdk.admin.uploads
        .create({ files })
        .catch(() => {
          form.setError("media", {
            type: "invalid_file",
            message: t("products.media.failedToUpload"),
          })
        })
```

**Review by adrien2p — APPROVED** [review]

File: packages/modules/providers/file-s3/src/services/s3-file.ts (hunk 1 above)
Comment (sradevski) [reply packages/modules/providers/file-s3/src/services/s3-file.ts:102]: @carlos-r-l-rodrigues we haven't added support for private files yet, so I'm leaving it hard-coded until we do.

File: packages/admin-next/dashboard/src/routes/products/product-media/components/edit-product-media-form/edit-product-media-form.tsx (hunk 2 above)
Comment (sradevski) [reply packages/admin-next/dashboard/src/routes/products/product-media/components/edit-product-media-form/edit-product-media-form.tsx:99]: 👍 I'll open another PR to add upload support when creating a product, and I'll refactor it then anyway

**Pull request merged**